### PR TITLE
Managing saved searches for other users

### PR DESCRIPTION
**Pull request by tomaskikutis**
SDFID-475

DEPENDS ON: SDESK-3322

### CODE REVIEW

**Comment by tomaskikutis** [comment]
@fritzSF don't forget the icon for "desk avatar"

**Comment by tomaskikutis** [comment]
@jerome-poisson ping me when you are done fixing the API

**Comment by jerome-poisson** [comment]
@tomaskikutis sorry, forgot to ping you, should be alright now.

File: scripts/apps/search/directives/EditTimeInterval.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+EditTimeInterval.$inject = ['asset'];
+
+export function EditTimeInterval(asset) {
+    return {
+        scope: {
+            initialValue: '=',
+            onChange: '=',
+        },
+        templateUrl: asset.templateUrl('apps/search/views/edit-time-interval.html'),
```
Comment (petrjasek) [inline scripts/apps/search/directives/EditTimeInterval.ts]: you can use `template: require('../views/edit-time-interval.html')`

**Comment by petrjasek** [comment]
just wondering, is the `business-logic` folder some kind of convention with ts? imo it's a bit confusing, I would look there for services and not types/interfaces

**Comment by tomaskikutis** [comment]
> just wondering, is the business-logic folder some kind of convention with ts?

Nope, that's what I came up with at the moment. Does core-interfaces sound better? I want such folder to be able to easily see and understand the core structures of the system. I'd like to have schema there, interfaces of services and operations between them.

I would like to avoid putting implementation specific stuff there like interfaces of directives/controllers or react components since they are low-level implementation details.

**Comment by petrjasek** [comment]
I wouldn't use the core prefix, `interfaces` sounds good to me

**Comment by petrjasek** [comment]
or `types`

**Comment by tomaskikutis** [comment]
As I mentioned, I think it would be easier to understand the system if we don't put low-level implementation types there and I want that to be clear from the naming.

File: scripts/apps/search/directives/EditTimeInterval.ts (hunk 1 above)
Comment (petrjasek) [reply scripts/apps/search/directives/EditTimeInterval.ts]: yep that's right. we can also make require lazy via webpack config, but atm it's bundled together

**Comment by petrjasek** [comment]
I see, imo core doesn't add much info there, maybe `superdesk-interfaces` or `superdesk-types` ?

File: scripts/apps/search/views/saved-search-subscribe.html [hunk 2]
```diff
@@ -7,39 +7,27 @@ <h3 class="modal__heading" translate="">
         </h3>
     </div>
 
-    <div class="modal__body" style="min-height: 550px;">
+    <div class="modal__body">
         <div class="sd-alert sd-alert--hollow sd-alert--primary">
             <p>You have chosen to subscribe to <strong>{{savedSearch.name}}</strong>. Please define the frequency and time of your email notifications below.</p>
```
Comment (petrjasek) [inline scripts/apps/search/views/saved-search-subscribe.html]: missing translate

File: scripts/apps/search/views/search-panel.html [hunk 3]
```diff
@@ -22,6 +22,23 @@
...
 
         <div id="edit-search-name" class="edit-search" ng-show="editingSearch && !activateSearchPane">
+            <div
+                ng-if="userHasPrivileges({saved_searches_subscriptions_admin: 1})"
+                class="saved-search-subscriptions--manage-toolbar">
+                <span>
+                    <strong>
+                        {{wrapper.edit.subscribers == null ? '0' : wrapper.edit.subscribers.user_subscriptions.length}}
+                    </strong>
+                    subscriptions
```
Comment (petrjasek) [inline scripts/apps/search/views/search-panel.html]: missing translate

File: scripts/apps/search/views/saved-search-manage-subscribers.html [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        </h3>
+    </div>
+
+    <div class="modal__body">
+        <!-- if start -->
+        <div ng-if="wrapper.subscriptionInCreateOrEditMode == null">
+            <div class="space-between" style="padding-bottom: 2rem; color: #898989">
+                <div>
+                    <strong>{{getSubscriptionCount()}}</strong>
+                    <span translate>subscriptionss</span>
```
Comment (petrjasek) [inline scripts/apps/search/views/saved-search-manage-subscribers.html]: typo `s/ss</s</`

File: scripts/apps/search/views/edit-time-interval.html [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+</div>
+
+<div
+    ng-if="wrapper.everyDay !== 'true'"
+    sd-weekday-picker
+    data-short-weekday-labels="true"
+    data-model="wrapper.customWeekdays"
+    class="form__row"></div>
+
+<div class="form-label">Time</div>
```
Comment (petrjasek) [inline scripts/apps/search/views/edit-time-interval.html]: missing translate

File: scripts/apps/search/views/edit-time-interval.html [hunk 6]
```diff
@@ -0,0 +1,27 @@
+<div class="form-label">Schedule</div>
```
Comment (petrjasek) [inline scripts/apps/search/views/edit-time-interval.html]: missing translate

File: scripts/apps/search/views/edit-time-interval.html [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+<div class="form__row">
+    <sd-check type="radio" ng-model="wrapper.everyDay" ng-value="true" label-position="inside">Every day</sd-check>
```
Comment (petrjasek) [inline scripts/apps/search/views/edit-time-interval.html]: missing translate

File: scripts/apps/search/views/edit-time-interval.html [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    <sd-check type="radio" ng-model="wrapper.everyDay" ng-value="false" label-position="inside">Select days</sd-check>
```
Comment (petrjasek) [inline scripts/apps/search/views/edit-time-interval.html]: missing translate

File: scripts/apps/search/views/edit-time-interval.html [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    ng-if="wrapper.everyDay !== 'true'"
+    sd-weekday-picker
+    data-short-weekday-labels="true"
+    data-model="wrapper.customWeekdays"
+    class="form__row"></div>
+
+<div class="form-label">Time</div>
+
+<div class="form__row">
+    <sd-check type="radio" ng-model="wrapper.everyHour" ng-value="true" label-position="inside">Every hour</sd-check>
```
Comment (petrjasek) [inline scripts/apps/search/views/edit-time-interval.html]: missing translate

File: scripts/apps/search/views/edit-time-interval.html [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    <sd-check type="radio" ng-model="wrapper.everyHour" ng-value="false" label-position="inside">Select hours</sd-check>
```
Comment (petrjasek) [inline scripts/apps/search/views/edit-time-interval.html]: missing translate

File: scripts/apps/search/views/saved-search-manage-subscribers.html (hunk 4 above)
Comment (tomaskikutis) [reply scripts/apps/search/views/saved-search-manage-subscribers.html]: I'm not sure I understand your formatting :D - `s/ss</s</`. I see that there's an extra `s` in `<span translate>subscriptionss</span>`. Do you see more typos in this file?

**Pull request merged**